edx_repo_tools/find_dependencies/find_python_dependencies.py

```diff
@@ -0,0 +1,90 @@
+"""
+Spider and catalog dependencies.
+$ python find_python_dependencies.py $FILE_PATH
+"""
+
+import click
+import json
+import os
+import requirements
+import sys
+from pathlib import Path
+import requests
+
+
+# The first of these we find is the requirements file we'll examine:
+PY_REQS = [
+    "requirements/edx/base.txt",
+    "requirements/base.txt",
+    "requirements.txt",
+]
+
+def request_package_info_url(package):
+        base_url = "https://pypi.org/pypi/"
+        url = f"{base_url}{package}/json"
+        response = requests.get(url)
+        if response.status_code == 200:
+            data_dict = response.json()
+            info = data_dict["info"]
+            return info["home_page"]            
+        else:
+            print(f"Failed to retrieve data for package {package}. Status code:", response.status_code)    
+
+FIRST_PARTY_ORGS = ["openedx"]
+
+SECOND_PARTY_ORGS = [
+    "edx", "edx-unsupported", "edx-solutions",
+    "mitodl",
+    "overhangio",
+    "open-craft", "eduNEXT", "raccoongang",
+]
+
+def urls_in_orgs(urls, orgs):
+    """
+    Find urls that are in any of the `orgs`.
+    """
+    return sorted(
+        url for url in urls
+        if any(f"/{org}/" in url for org in orgs)
+    )
+
+@click.command()
+@click.option(
+    '--dirs', 'directories',
+    multiple=True,
+    help="Directories path to find the python dependencies. The path should be "
+         "absolute to a particular repository. "
+         "This option can be provided multiple times.",
+)
```

Comment:
Let's make this option `--req-file` and update the help comment so it's more accurate.

Reply:
done